terraform/database/main.tf

```diff
@@ -0,0 +1,47 @@
+###################################################################################
+
+# Creates a Cloud SQL instance, database, and user. 
+# Returns a database_url connection string, and database_instance triple.
+
+###################################################################################
+
+locals {
+  database_user = "${var.slug}-django-user"
+  database_name = "${var.slug}-database"
+
+  database_instance_fqdn = "${var.project}:${var.region}:${google_sql_database_instance.postgres.name}"
+}
+
+###################################################################################
+
+resource "random_password" "database_user_password" {
+  length  = 30
+  special = false
+}
+
+resource "google_sql_database_instance" "postgres" {
+  name             = var.instance_name
+  database_version = "POSTGRES_11"
+  region           = var.region
+
+  settings {
+    tier = "db-f1-micro"
+  }
+}
+
+resource "google_sql_database" "database" {
+  name     = local.database_name
+  instance = google_sql_database_instance.postgres.name
+}
+
+
+# TODO(glasnt) - Current database user has cloudsqladmin rights. Requires reducing. 
+resource "google_sql_user" "user" {
+  name     = local.database_user
+  instance = google_sql_database_instance.postgres.name
+  password = random_password.database_user_password.result
```

Comment:
Be careful with random_password and SQL passwords. I've had some of the generated passwords rejected for odd reasons.

Reply:
I added the `special = false` for this reason, but I totally understand the worry.